pkg/controller/controller_test.go

```diff
@@ -369,13 +370,125 @@ func TestReconcileBroker(t *testing.T) {
 	}
 
 	// second action should be an update action for broker status subresource
-	createAction2 := actions[1].(clientgotesting.CreateAction)
-	if e, a := "update", createAction2.GetVerb(); e != a {
+	updateAction := actions[1].(clientgotesting.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
 		t.Fatalf("Unexpected verb on actions[1]; expected %v, got %v", e, a)
 	}
 
-	createActionObject2 := createAction2.GetObject().(*v1alpha1.Broker)
-	if e, a := testBrokerName, createActionObject2.Name; e != a {
+	updateObject := updateAction.GetObject().(*v1alpha1.Broker)
+	if e, a := testBrokerName, updateObject.Name; e != a {
+		t.Fatalf("Unexpected name of broker created: expected %v, got %v", e, a)
+	}
+
+	// verify no kube resources created
+	kubeActions := fakeKubeClient.Actions()
+	if e, a := 0, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+}
+
+func TestReconcileBrokerExistingServiceClass(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t)
+
+	testServiceClass := getTestServiceClass()
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(testServiceClass)
+
+	fakeBrokerClient.CatalogClient.RetCatalog = getTestCatalog()
+
+	testController.reconcileBroker(getTestBroker())
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 2, len(actions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v. Actions: %+v", e, a, actions)
+	}
+
+	// first action should be an update action for a service class
+	updateAction := actions[0].(clientgotesting.UpdateAction)
```

Comment:
I'm working on another PR that will check all of these assertions everywhere else in the file.  Are you down with that?  There are tons of places where assertions of this type are not checked in this file.